### ISSUE TITLE
default-reporter: show a list of workspace packages matching the filter

### ISSUE THREAD

**Issue by idan-at** [pnpm/pnpm#3038]
Currently, when running `pnpm recursive test --filter ...[origin/master]`, the output is: `Scope: 38 of 132 workspace projects`.

I think it can be useful to add a list of all packages. for example, if loglevel is `silly, the output can look something like:
```
Scope: 38 of 132 workspace projects:

project1
nested/project2
...
```

wdyt?

**Comment by zkochan** [comment]
but the path to each package is printed anyway when their script is executed.

**Comment by idan-at** [comment]
true, but it's very hard to understand in retrospect which packages ran, mostly because:
- a command can run in parallel, making the log harder to track
- the log can be huge :)

Originally I used `pnpm recursive exec pwd --filter ${filter}` to get the list of packages that will run the given command, but after we've added `test-pattern` this won't be correct (since `test` is a special case)

**Comment by zkochan** [comment]
`pnpm recursive exec pwd --filter ${filter}` this will work. I've made `--test-pattern` work with all commands to avoid confusion.
